include/hipSYCL/sycl/libkernel/host/group_functions.hpp

```diff
@@ -388,12 +388,22 @@ inline void group_barrier(Group g, memory_scope fence_scope = Group::fence_scope
     g.barrier();
```

Comment:
btw. there's also system scope.. should at least do a `g.barrier()` there, too? Maybe add a `switch` without default, so this (hopefully) generates a warning if enum values are missing?
https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_memory_scope

Reply:
hm, this looks fishy indeed - don't we need a barrier in all cases if the `Group` is a work group, regardless of the fence scope?
If the group is a work group, you would always expect work group barrier semantics.

Reply:
yes, we always need a barrier and a memory-fence for the device-scope, no idea how I messed that up :/